internal/loader/configuration_setting_loader.go

```diff
@@ -471,37 +579,52 @@ func getSecret(ctx context.Context, namespacedSecretName types.NamespacedName) (
 	return secretObject, nil
 }
 
-func getFilters(acpSpec acpv1.AzureAppConfigurationProviderSpec) []acpv1.KeyValueSelector {
-	var filters []acpv1.KeyValueSelector
+func getFilters(acpSpec acpv1.AzureAppConfigurationProviderSpec) ([]acpv1.KeyValueSelector, []acpv1.KeyValueSelector) {
```

Comment:
Don't use one method to get both `KeyValueSelector` and `KeyValueSelector`. Split to two methods